spacy_experimental/span_boundary_detection/sbd_suggester.py

```diff
@@ -0,0 +1,54 @@
+from typing import Optional, Iterable, cast
+from thinc.api import get_current_ops, Ops
+
+from thinc.types import Ragged, Ints1d
+
+from spacy.compat import Protocol, runtime_checkable
+from spacy.tokens import Doc
+from spacy.util import registry
+
+
+@runtime_checkable
+class Suggester(Protocol):
+    def __call__(self, docs: Iterable[Doc], *, ops: Optional[Ops] = None) -> Ragged:
+        ...
+
+
+@registry.misc("spacy-experimental.span_boundary_detection_suggester.v1")
+def build_sbd_suggester() -> Suggester:
+    """Suggest every candidate predicted by the SBD"""
+
+    def sbd_suggester(docs: Iterable[Doc], *, ops: Optional[Ops] = None) -> Ragged:
+        if ops is None:
+            ops = get_current_ops()
+        spans = []
+        lengths = []
+        for doc in docs:
+            starts = []
+            ends = []
+            length = 0
+            cache = {}
+            for token in doc:
+                if token._.span_start == 1:
+                    starts.append(token.i)
+                if token._.span_end == 1:
+                    ends.append(token.i + 1)
+
+            for start in starts:
+                for end in ends:
+                    if start < end and (start, end) not in cache:
+                        spans.append([start, end])
+                        cache[(start, end)] = True
```

Comment:
the `cache` here could just be a `set()` and then `if (start, end) not in cache` still makes sense.

Reply:
That's so embarrassing, how did I not know of `sets` before? I'm using this workaround with `dicts` all the TIME! This is super useful 😆